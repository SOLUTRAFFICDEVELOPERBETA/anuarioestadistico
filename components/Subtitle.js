import React from 'react'
import { Typography, Divider, Box } from '@material-ui/core'
import PropTypes from 'prop-types'

/**
 * @description Componente que permite integrar un subtitulo en un pÃ¡gina
 * @param {String} primary 
 * @param {String} secondary 
 * Consulte los elementos utilizados en {@link https://material-ui.com/ | Material-ui}
 */
const Subtitle = ({ primary, secondary }) => (
  <Box marginBottom={2}>
    <Typography variant="h6" color="primary" style={{ textTransform: 'uppercase' }}>
      {primary}
    </Typography>
    <Typography variant="body2" color="secondary">
      <em>{secondary}</em>
    </Typography>
    <Divider variant="inset" />
  </Box>
)

Subtitle.propTypes = {
  primary: PropTypes.string.isRequired,
  secondary: PropTypes.string
}

Subtitle.defaultProps = {
  secondary: ''
}

export default Subtitle
