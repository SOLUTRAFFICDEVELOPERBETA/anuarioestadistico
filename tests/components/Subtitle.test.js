import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import Subtitle from '../../components/Subtitle';

describe('Pruebas en el componente <Subtitle/>', () => {
   
    const wrapper = shallow(
        <Subtitle
            primary={"Texto primario de prueba"}
            secondary={"Texto secundario de prueba"}
        />

    )

    test('debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('debe llamar la funciÃ³n onOver()', () => {
        expect(wrapper.find('WithStyles(ForwardRef(Typography))').at(0).text().trim()).toBe('Texto primario de prueba')
        expect(wrapper.find('WithStyles(ForwardRef(Typography))').at(1).text().trim()).toBe('Texto secundario de prueba')
    })


})