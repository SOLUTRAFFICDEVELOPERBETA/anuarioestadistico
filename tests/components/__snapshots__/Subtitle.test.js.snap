// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <Subtitle/> debe mostrarse correctamente 1`] = `
<Styled(MuiBox)
  marginBottom={2}
>
  <WithStyles(ForwardRef(Typography))
    color="primary"
    style={
      Object {
        "textTransform": "uppercase",
      }
    }
    variant="h6"
  >
    Texto primario de prueba
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Typography))
    color="secondary"
    variant="body2"
  >
    <em>
      Texto secundario de prueba
    </em>
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Divider))
    variant="inset"
  />
</Styled(MuiBox)>
`;
